### PR TITLE
Update Semantic Search with Hugging Face notebook

### PR DESCRIPTION
**Pull request by tclendaniel-singlestore**
- Pin sentence-transformers/transformers to versions the hosted image supports, and assert the PyTorch backend initialized
- Store embeddings in a native VECTOR(384, F32) column via an explicit CREATE TABLE and UNHEX insert, instead of an inferred BLOB schema
- Normalize embeddings so DOT_PRODUCT is a cosine similarity, and use the <*> operator in the search query
- Derive the vector dimension from the loaded model
- Expand the markdown to explain keyword vs semantic search, mean pooling, the 128-token limit, and vector indexes
- Print search results as readable scored excerpts

---

> [!NOTE]
> **Low Risk**
> Documentation and demo-notebook-only changes; no production services or shared libraries affected.
> 
> **Overview**
> **Refreshes the Semantic Search with Hugging Face notebook** so it matches current hosted-image dependencies, native SingleStore vectors, and clearer teaching narrative.
> 
> **Runtime and embeddings:** Dependency install moves to bounded `sentence-transformers` / `transformers` versions (drops pinned torch/tensorflow/datasets stack), prints versions, and **asserts the PyTorch backend is active**. Loading uses **`SentenceTransformer`** with **`normalize_embeddings=True`** and batched `encode()`; **`EMBEDDING_DIM`** is inferred from the model for schema alignment. IMDB data is loaded via **`hf_hub_download` + `test.jsonl`** with a fixed **`random_state`** sample.
> 
> **Storage and search:** Replaces **`pandas.to_sql` + BLOB inference** with an explicit **`VECTOR(N, F32) NOT NULL`** table and **`UNHEX`** inserts for packed float32 bytes. Search SQL switches to the **`<*>`** infix dot-product operator with **`UNHEX`**, and results print as **scored excerpts**.
> 
> **Docs:** Intro and step markdown are expanded (keyword vs semantic search, mean pooling, 128-token truncation, optional IVF vector index) plus kernel-restart and long-review notes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6958e34d3c228e0323ab1ce99a97ead108302599. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: notebooks/semantic-search-with-hugging-face/notebook.ipynb [hunk 1]
```diff
@@ -34,11 +34,20 @@
...
+        "In this notebook you will build a working semantic search engine over movie reviews: Hugging Face turns text into vectors, and SingleStore stores and searches them with a single line of SQL.\n",
         "\n",
-        "SingleStoreDB\u2019s built-in parallelization and Intel SIMD-based vector processing takes care of the heavy lifting involved in processing vector data. This allows your to run your ML algorithms right in your database extremely efficiently with just 1 line of SQL!\n",
+        "**Keyword search** matches the characters you typed. If a review says \"hilarious\" and you search for \"funny\", keyword search finds nothing. **Semantic search** matches meaning instead. An embedding model converts each piece of text into a list of numbers \u2014 a *vector* \u2014 positioned so that texts with similar meanings land near each other. Finding relevant reviews then becomes a geometry question: which stored vectors point in most nearly the same direction as the vector for your query?\n",
```
Comment (s2-ktran) [inline notebooks/semantic-search-with-hugging-face/notebook.ipynb:39]: Really like how you included this example!

File: notebooks/semantic-search-with-hugging-face/notebook.ipynb [hunk 2]
```diff
@@ -108,28 +138,49 @@
...
     },
     {
       "cell_type": "markdown",
       "metadata": {},
       "source": [
-        "## 4. Load Sentence Transformer library and create a function called `get_embedding()`\n",
+        "## 4. Load the embedding model and create a `get_embedding()` function\n",
+        "\n",
+        "We use `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`: a small, fast model\n",
+        "trained so that sentences which paraphrase one another sit close together in vector space,\n",
```
Comment (s2-ktran) [inline notebooks/semantic-search-with-hugging-face/notebook.ipynb:176]: Like the added detail.

**Review by s2-ktran — APPROVED** [review]
Great changes for explaining what's happening in the notebook! Excited to see a video walkthrough on this one.
